src/button/styled-components.js

```diff
@@ -29,52 +38,62 @@ export const BaseButton = styled(
       backgroundColor: $theme.colors.mono300,
       color: $theme.colors.mono600,
     },
-    // Padding
-    ...($shape === SHAPE.round || $shape === SHAPE.square
-      ? {
-          paddingTop:
-            $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
-          paddingBottom:
-            $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
-          paddingLeft:
-            $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
-          paddingRight:
-            $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
-        }
-      : {
-          paddingTop:
-            $theme.sizing[$size === SIZE.compact ? 'scale200' : 'scale300'],
-          paddingBottom:
-            $theme.sizing[$size === SIZE.compact ? 'scale200' : 'scale300'],
-          paddingLeft: $theme.sizing.scale600,
-          paddingRight: $theme.sizing.scale600,
-        }),
+    // Padding For Shape and Size
+    ...getStyleForShape({$theme, $shape, $size}),
     // Kind style override
-    ...getStyleForKind({$theme, $kind}),
+    ...getStyleForKind({$theme, $kind, $isLoading}),
   }),
 );
 
-export const EndEnhancer = styled('div', ({$theme}) => ({
+export const EndEnhancer = styled('div', ({$theme}: StylePropsT) => ({
   display: 'flex',
   marginLeft: $theme.sizing.scale500,
 }));
 
-export const StartEnhancer = styled('div', ({$theme}) => ({
+export const StartEnhancer = styled('div', ({$theme}: StylePropsT) => ({
   display: 'flex',
   marginRight: $theme.sizing.scale500,
 }));
 
-function getStyleForKind({$theme, $kind}) {
+export function getStyleForShape({$theme, $shape, $size}: StylePropsT) {
+  switch ($shape) {
+    case SHAPE.round:
+    case SHAPE.square:
+      return {
+        paddingTop:
+          $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
```

Comment:
this notation is better for some IDEs as allows indexing and search better
```js
$size === SIZE.compact ? $theme.sizing.scale400 : $theme.sizing.scale500
```

Reply:
didn't think about that, happy to switch!